dependencies/meson.build

```diff
@@ -0,0 +1,48 @@
+# nix dependencies build file
+#============================================================================
+
+
+# declare dependency types
+#---------------------------------------------------
+required_dep_dirs = [
+    'archive',
+    'boost',
+    'brotli',
+    'bz2',
+    'cpuid',
+    'curl',
+    'dl',
+    'editline',
+    'lzma',
+    'lowdown',
+    'nlohmann',
+    'openssl',
+    'seccomp',
+    'sqlite',
+    'pthread',
+    'zlib']
+
+optional_dep_dirs = [
+    'aws-sdk-cpp',
+    'bdw-gc',
+    'gtest',
+    'sodium']
+
+check_dirs = [
+    'functions',
+    'headers']
+
+
+# Build dependencies
+#---------------------------------------------------
+foreach dir : required_dep_dirs
```

Comment:
Could be using a list concatenation to simplify duplication here:
```
foreach dir : required_dep_dirs + optional_dep_dirs + check_dirs
    subdir(dir)
endforeach
```

Reply:
looks clunky and effects readability imo